rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -3109,20 +3109,21 @@ rmw_destroy_wait_set(rmw_wait_set_t * wait_set)
   return RMW_RET_OK;
 }
 
+//==============================================================================
 static bool has_triggered_condition(
-  rmw_subscriptions_t * subscriptions,
-  rmw_guard_conditions_t * guard_conditions,
-  rmw_services_t * services,
-  rmw_clients_t * clients,
-  rmw_events_t * events)
+  const rmw_subscriptions_t * const subscriptions,
+  const rmw_guard_conditions_t * const guard_conditions,
+  const rmw_services_t * const services,
+  const rmw_clients_t * const clients,
+  const rmw_events_t * const events)
 {
   static_cast<void>(events);
 
   if (guard_conditions) {
     for (size_t i = 0; i < guard_conditions->guard_condition_count; ++i) {
       GuardCondition * gc = static_cast<GuardCondition *>(guard_conditions->guard_conditions[i]);
       if (gc != nullptr) {
-        if (gc->get_and_reset_trigger()) {
+        if (gc->get_trigger()) {
```

Comment:
Just for posterity.  What I _think_ was happening here is that checking for triggered conditions wasn't idempotent and was actually mutating the underlying conditions.  Now we are only _reading_ the condition (and not resetting the trigger) each time we are checking.

Reply:
> Just for posterity. What I _think_ was happening here is that checking for triggered conditions wasn't idempotent and was actually mutating the underlying conditions. Now we are only _reading_ the condition (and not resetting the trigger) each time we are checking.

Yep, you are exactly right.

Reply:
Just as an FYI, I've come up with a completely different approach to implementing `rmw_wait` on https://github.com/ros2/rmw_zenoh/tree/clalancette/rewrite-rmw-wait .  It needs more work, but I also think that it has  more promise than this PR to be much more performant.  I haven't opened a PR yet since it is still WIP.